src/main/java/com/service/UserServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.model.User;
+import com.repository.UserRespository;
+
+@Service
+public class UserServiceImpl implements UserService {
+    
+    private UserRespository userRepo;
+
+    @Autowired
+    public void UserService(UserRespository userRepo) {
+        this.userRepo = userRepo;
+    }
+
+    public Iterable<User> getAllUsers() {
+        return userRepo.findAll();
+    }
+
+    public User addUser(User user) {
+        return userRepo.save(user);
+    }
+
+
+    // delete
+    public String deleteUser(String id) {
+        userRepo.deleteById(id);
+        return "User Deleted";
+    }
+
+    // update
+    public User updateUser(User user) {
+        return userRepo.save(user);
+    }
+
+    // update user information
+    public User updateUser(String id, User user) {
```

Comment:
it is impossible for users to send a request with the user type
try to convert it on your own
also the id should be in ObjectId type